src/image/loading_displaying.js

```diff
@@ -1110,7 +1213,20 @@ p5.prototype.image = function(
   // set defaults per spec: https://goo.gl/3ykfOq
 
   p5._validateParameters('image', arguments);
-
+  // If dz is not specified.
+  if (arguments.length % 2 === 1 ||  typeof arguments[9]  === 'string'){
```

Comment:
I'm not completely sure if this is causing the test to fail, but I believe the code enters the if block in both cases—whether dz is specified or not.

For example, when you check arguments.length, it will be 11 if z is provided (which is an odd number), and arguments[9] will be a number.

However, if the dz variable isn't specified, arguments.length will be 10 (even), and in this case, arguments[9] will be a string.

If you do something like:
```
if(arg.length%2 !== 1 || typeof arg[9] ==='string){...}
``` 

is that works for you?

Reply:
Also, I’m concerned that we might need a different approach for other cases. For example, if we have a function call like `image(img, 100, 100, 0);`, the number of arguments is four, we have `arg.length === 1` for which z is not specified. 

We could probably have a general case for all the cases which could work for image(no. of arguments).

let me know what you think?